packages/graalvm/src/main/kotlin/elide/runtime/gvm/internals/node/fs/NodeFilesystem.kt

```diff
@@ -303,8 +341,6 @@ internal abstract class FilesystemBase (
   }
 
   protected inline fun <reified T> openAndRead(path: java.nio.file.Path, op: SeekableByteChannel.() -> T): T {
-    checkFileForRead(path)
-
```

Comment:
this should eliminate a few syscalls which are HOL blocking and shouldn't be here anyway; if a file is not readable or does not exist, we should surface an exception and fail.